### PR TITLE
Fix timing of discarding plot to used pile

### LINKED ISSUE #2125: Sand Steed timing
Sand Steed power should trigger after when revealed effects have been resolved. 

Currently creating some odd situations such as being able to trigger when opponent is revealing king in the north, or letting someone trigger and win a game before a valar morghulis, when actually the characters should die first.

### PR DESCRIPTION
**Pull request by ystros**
Fixes #2125

### CODE REVIEW

File: server/game/GameActions/LeavePlay.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    createEvent({ card, allowSave = false }) {
+        const params = {
+            card,
+            allowSave,
+            automaticSaveWithDupe: true
+        };
+
+        return this.event('onCardLeftPlay', params, event => {
+            event.cardStateWhenLeftPlay = event.card.createSnapshot();
```
Comment (EvadLhorg) [inline server/game/GameActions/LeavePlay.js:22]: IIRC we wanted to seperate the snapshot creation from the event handler like we did for `DiscardCard.js` so interrupts that overwrite the handler don´t overwrite the snapshot creation. I suggest adding `snapshotName: 'cardStateWhenLeftPlay'` to the params and then the snapshot will be created automatically inside `event.js`

File: server/game/gamesteps/plotphase.js [hunk 2]
```diff
@@ -51,27 +49,15 @@ class PlotPhase extends Phase {
...
-        }
-    }
-
     recyclePlots() {
         for(const player of this.game.getPlayers()) {
             player.recyclePlots();
         }
     }
 
     getActivePlots() {
```
Comment (EvadLhorg) [inline server/game/gamesteps/plotphase.js:58]: maybe rename to `getSelectedPlots()` as the plots are not active yet?

**Review by EvadLhorg — CHANGES_REQUESTED** [review]

File: server/game/GameActions/LeavePlay.js (hunk 1 above)
Comment (ystros) [reply server/game/GameActions/LeavePlay.js]: Good catch, this class was originally put together before you had added that mechanism.

File: server/game/gamesteps/plotphase.js (hunk 2 above)
Comment (ystros) [reply server/game/gamesteps/plotphase.js]: Yup. I want to totally change how plot selection is done so this method will go away eventually, but for now, the name will be updated.

**Pull request merged**